### ISSUE TITLE
[rke2] Windows unable to pull registration command script in a proxied environment

### ISSUE THREAD

**Issue by slickwarren** [rancher/rancher#36574]
**Rancher Server Setup**
- Rancher version: v2.6.4-rc3
- Installation option (Docker install/Helm Chart): helm
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): rke1
- Proxy/Cert Details: self signed

**Information about the Cluster**
- Kubernetes version: 1.21.9/1.22.6 (rke2)
- Cluster Type (Local/Downstream): downstream
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider): custom + windows

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom): any

   
**Describe the bug**

when registering a windows node in a proxied setup (proxied rancher, proxied windows node) windows is unable to register. The script does not start to pull from rancher. 

**To Reproduce**
 
* deploy rancher in a proxied environment
* deploy a proxied downstream custom cluster
  * register windows worker node in the cluster

**Result**
* windows node is able to curl the rancher URL/other public URLs, however can't pull the script to register with the cluster. 

**Expected Result**

windows should be able to pull the script in a proxied environment 

**Additional context**
* followed [these instructions](https://github.com/rosskirkpat/docs/blob/main/rke2/rke2-windows-troubleshooting-qa.md#how-to-check-named-pipes) on windows node to get proxy setup
* not sure if this will root cause as a rancher or windows issue. 
* setup available for debugging upon request

**Comment by slickwarren** [comment]
After bringing up another environment, windows team was able to find and debug the issue. @rosskirkpat will file the underlying bug(s) shortly.

**Comment by rosskirkpat** [comment]
@slickwarren Can you provide the Jenkins job(s) that you used to setup the environment?

**Comment by sirredbeard** [comment]
This issue is very likely a duplicate and has already been QA'd.

**Comment by slickwarren** [comment]
tested using v2.6.5-rc1:
the reg. command is able to finish pulling and runs the script. However, it hangs just before running rke2. From the rancher-wins logs:
```
Message : Pulling image index.docker.io/rancher/system-agent-installer-rke2:v1.22.8-rke2r1

Message : Using private registry config file at /etc/rancher/agent/registries.yaml

Message : [Applyinator] Extracting image rancher/system-agent-installer-rke2:v1.22.8-rke2r1 to directory
          C:\var\lib\rancher\agent\work\20220418-221552\0ac288cdd81de8b67006f5dd3d6e0f79beba9fc0d24e9e3d71d590f5ea3905f0_0

Message : [Applyinator] Applying one-time instructions for plan with checksum 0ac288cdd81de8b67006f5dd3d6e0f79beba9fc0d24e9e3d71d590f5ea3905f0

Message : Detected first start, force-applying one-time instruction set

Message : Starting /v1, Kind=Secret controller

Message : Listening on \\.\pipe\rancher_wins

Message : Starting remote watch of plans

Message : Setting C:/var/lib/rancher/agent/work as the working directory

Message : Listening on \\.\pipe\rancher_wins_proxy

Message : Rancher System Agent version dev (HEAD) is starting

Message : Stackdump - waiting signal at Global\stackdump-3468
```
reopening for now. Please ping me as I still have a setup around for testing.

**Comment by slickwarren** [comment]
same issue as https://github.com/rancher/rancher/issues/36552#issuecomment-1110315090, blocking on this ticket

**Comment by rosskirkpat** [comment]
Once https://github.com/rancher/rancher/pull/37513 is merged and present in an RC, this is ready to be tested again.

**Comment by rosskirkpat** [comment]
@luthermonson is fixing a bug in the wins config file for whitelisting, which is requiring a v0.2.5 release. He will be coordinating a new PR and attach it to this issue.

**Comment by rosskirkpat** [comment]
https://github.com/rancher/rancher/pull/37527 is merged. Moving this to waiting on Rancher RC

**Comment by slickwarren** [comment]
tested on v2.6.5-rc6 - when registering the windows rke2 node (1.22.9+rke2r1) the cluster gives the following error: error applying plan -- check rancher-system-agent.service logs on node for more information, waiting for agent to check in and apply initial plan

Ross is currently investigating the issue.

**Comment by sirredbeard** [comment]
Will be fixed by PR https://github.com/rancher/rancher/pull/37586

**Comment by slickwarren** [comment]
using rancher v2.6.5-rc8 there is an issue when trying to connect the windows node to rancher 
```
Start-Service : Service 'Rancher Wins (rancher-wins)' cannot be started due to the following error: Cannot start service rancher-wins on computer '.'.
```
reopening for this issue (Ross is aware of the issue and is working on a fix currently)

**Comment by rosskirkpat** [comment]
This issue is ready to be tested again. There were two fixes that went in on Friday, one fixed a nil pointer in system-agent specific to windows and the other was improving the logic in wins for csi-proxy and tlsConfig

**Comment by slickwarren** [comment]
there was an issue found on v2.6.5-rc10 however the full fix won't be able to make it in 2.6.5. We have the following PRs:
https://github.com/rancher/rancher/issues/37688
https://github.com/rancher/rancher/pull/37684

that have been merged and will be part of 2.6.5 so that the OOB fix will work.

**Comment by sirredbeard** [comment]
With the latest releases, this should go back to test.

**Comment by slickwarren** [comment]
tested on 2.6-head (c8d01c6) using k8s 1.23.6+rke2r2 and just realized that KDM has not yet been released to test this yet. Wednesday is the release for kdm.
